### PR TITLE
[Fix clojure-emacs/cider#1323] Sexp navigation near end of buffer

### PR DESCRIPTION
**Pull request by Malabarba**

### CODE REVIEW

File: clojure-mode.el [hunk 1]
```diff
@@ -1087,8 +1088,7 @@ Returns a list pair, e.g. (\"defn\" \"abc\") or (\"deftest\" \"some-test\")."
 (defun clojure--looking-at-logical-sexp ()
   "Return non-nil if sexp after point represents code.
 Sexps that don't represent code are ^metadata or #reader.macros."
-  (forward-sexp 1)
-  (forward-sexp -1)
+  (comment-forward (point-max))
```
Comment (expez) [inline clojure-mode.el]: Should you be calling `comment-normalize-vars` first here?

Comment (Malabarba) [inline clojure-mode.el]: Good catch. Fixed.

**Comment by bbatsov** [comment]
I guess more changes would be needed.

**Comment by Malabarba** [comment]
What do you mean?

**Comment by expez** [comment]
@Malabarba the tests are failing

**Comment by Malabarba** [comment]
Oh. :)

**Comment by Malabarba** [comment]
Ah, it's some weird interaction with the current indentation engine. It's going to fix itself when we merge the other PR, so we can just sit on this one until then. :)

**Comment by bbatsov** [comment]
OK

**Comment by bbatsov** [comment]
Can you, please, rebase this?

**Comment by Malabarba** [comment]
Done. Added some tests too.

**Pull request merged**